paddle/fluid/operators/mkldnn/mul_mkldnn_op.cc

```diff
@@ -0,0 +1,537 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <string>
+#include <vector>
+#include "paddle/fluid/framework/data_layout_transform.h"
+#include "paddle/fluid/memory/malloc.h"
+#include "paddle/fluid/operators/mul_op.h"
+#include "paddle/fluid/platform/mkldnn_helper.h"
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+using framework::DataLayout;
+using Tensor = framework::Tensor;
+using framework::DataLayout;
+using framework::DDim;
+using framework::ExecutionContext;
+using framework::Tensor;
+using mkldnn::inner_product_forward;
+using mkldnn::memory;
+using mkldnn::primitive;
+using mkldnn::prop_kind;
+using mkldnn::stream;
+using platform::GetMKLDNNFormat;
+using platform::MKLDNNDeviceContext;
+using platform::to_void_cast;
+
+template <typename T, typename K>
+class MulPrimitiveFactory {
+ public:
+  explicit MulPrimitiveFactory(const mkldnn::engine& engine)
+      : engine_(engine) {}
+
+  virtual ~MulPrimitiveFactory() {}
+
+  virtual inner_product_forward CreateMulPrimitive(
+      const Tensor* input, const Tensor* weights, Tensor* output,
+      const ExecutionContext& ctx) {
+    /* check format and reorder if need */
+    int x_num_col_dims = ctx.Attr<int>("x_num_col_dims");
+    int y_num_col_dims = ctx.Attr<int>("y_num_col_dims");
+
+    memory::data_type x_dt = paddle::framework::ToMKLDNNDataType(input->type());
+    memory::data_type y_dt =
+        paddle::framework::ToMKLDNNDataType(weights->type());
```

Comment:
I think you don't need `paddle::` here

Reply:
okay, I'll fix it

Reply:
have you decided not to change this or not yet?

Reply:
Done! please refer to bb64cb8